pkg/cache/clusterqueue_test.go

```diff
@@ -223,58 +214,160 @@ func TestCohortCanFit(t *testing.T) {
+					Name: "C",
+					RequestableResources: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    5,
+							corev1.ResourceMemory: 5,
+						},
 					},
-				},
-				Usage: FlavorResourceQuantities{
-					"f1": map[corev1.ResourceName]int64{
-						corev1.ResourceCPU: 3,
+					Usage: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU:    5,
+							corev1.ResourceMemory: 5,
+						},
 					},
 				},
+				ResourceGroups: nil,
 			},
+		},
+		"missing resource": {
 			request: FlavorResourceQuantities{
 				"f1": map[corev1.ResourceName]int64{
 					corev1.ResourceCPU:    1,
 					corev1.ResourceMemory: 1,
 				},
 			},
 			wantFit: false,
+			cq: &ClusterQueue{
+				Name: "CQ",
+				Cohort: &Cohort{
+					Name: "C",
+					RequestableResources: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU: 5,
+						},
+					},
+					Usage: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU: 3,
+						},
+					},
+				},
+				ResourceGroups: nil,
+			},
+		},
+		"lendingLimit enabled can't fit": {
+			request: FlavorResourceQuantities{
+				"f1": map[corev1.ResourceName]int64{
+					corev1.ResourceCPU: 3,
+				},
+			},
+			wantFit: false,
+			cq: &ClusterQueue{
+				Name: "CQ-A",
+				Cohort: &Cohort{
+					Name: "C",
+					RequestableResources: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							// CQ-A has 2 nominal cpu, CQ-B has 3 nominal cpu and 2 lendingLimit,
+							// so when lendingLimit enabled, the cohort's RequestableResources is 4 cpu.
+							corev1.ResourceCPU: 4,
+						},
+					},
+					Usage: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							corev1.ResourceCPU: 2,
+						},
+					},
+				},
+				GuaranteedQuota: FlavorResourceQuantities{
+					"f1": {
+						corev1.ResourceCPU: 0,
+					},
+				},
+			},
+			enableLendingLimit: true,
+		},
+		"lendingLimit enabled can fit": {
+			request: FlavorResourceQuantities{
+				"f1": map[corev1.ResourceName]int64{
+					corev1.ResourceCPU: 3,
+				},
+			},
+			wantFit: true,
+			cq: &ClusterQueue{
+				Name: "CQ-A",
+				Cohort: &Cohort{
+					Name: "C",
+					RequestableResources: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							// CQ-A has 2 nominal cpu, CQ-B has 3 nominal cpu and 2 lendingLimit,
+							// so when lendingLimit enabled, the cohort's RequestableResources is 4 cpu.
+							corev1.ResourceCPU: 4,
+						},
+					},
+					Usage: FlavorResourceQuantities{
+						"f1": map[corev1.ResourceName]int64{
+							// CQ-B has admitted a workload with 2 cpus, but with 1 GuaranteedQuota,
+							// so when lendingLimit enabled, Cohort.Usage should be 2 - 1 = 1.
+							corev1.ResourceCPU: 1,
+						},
+					},
+				},
+				GuaranteedQuota: FlavorResourceQuantities{
+					"f1": {
+						corev1.ResourceCPU: 0,
```

Comment:
I think the guaranteedQuota now should be greater than 0, than we can cover different scenarios. That's say we should refactor these tests.